packages/rulesets/src/spectral/functions/properties-type-object-no-definition.ts

```diff
@@ -0,0 +1,15 @@
+// If a property is defined with `'type': 'object'`, it must also define an object body.
+// RPC Code: RPC-Policy-V1-03
+const errorMessage =
+  "If Properties with type:object dont have a reference model defined, then the allowed types can only be primitive data types."
+export const propertiesTypeObjectNoDefinition: any = (definitionObject: any, _opts: any, ctx: any) => {
+  if (definitionObject?.type === "") {
+    return [{ message: errorMessage, path: ctx.path }]
```

Comment:
>errorMessage [](http://example.com/codeflow?start=23&length=12)

Maybe throw a different error message for this case which says that the type MUST not be left empty?

Reply:
Okay I will add different error message.